### PR TITLE
Copy nodeaffinity predicate to its filter plugin

### LINKED ISSUE #86717: Move node Affinity predicate logic to its Filter plugin [Migration Phase 2]
/assign @ahg-g 
/sig scheduling
/priority important-soon

Part of #85822

### PR DESCRIPTION
**Pull request by ahg-g**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Copies node affinity logic to its Filter plugin. NodeAffinity predicate still exist because Kubelet and DaemonSet controller depends on it.

**Which issue(s) this PR fixes**:
Fixes #86717

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: pkg/scheduler/util/node_affinity.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+import (
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/labels"
+	api "k8s.io/kubernetes/pkg/apis/core"
+	v1helper "k8s.io/kubernetes/pkg/apis/core/v1/helper"
+)
+
+// TODO(#86236): move to a helper library.
```
Comment (ahg-g) [inline pkg/scheduler/util/node_affinity.go]: This file should ideally be in framework/plugins/helper, but placing it there right now will need updating to the dependency restrictions file of the controller package, which I prefer we do once in a separate PR (the scheduler/util package it already allowed there).

File: pkg/scheduler/util/node_affinity.go [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+import (
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/labels"
+	api "k8s.io/kubernetes/pkg/apis/core"
+	v1helper "k8s.io/kubernetes/pkg/apis/core/v1/helper"
+)
+
+// TODO(#86236): move to framework/plugins/helper package.
```
Comment (ahg-g) [inline pkg/scheduler/util/node_affinity.go]: This file should ideally be in framework/plugins/helper, but placing it there right now needs updating the dependency restrictions file of the controller package, which I prefer we do in a separate PR (the scheduler/util package it already allowed there). The reason I am moving it out of the predicates package is that I want to break the dependency of plugins to the predicates package.

**Review by Huang-Wei — COMMENTED** [review]
Thanks @ahg-g .

/lgtm

**Comment by ahg-g** [comment]
/hold
If #86730 gets merged first, then I can place PodMatchesNodeSelectorAndAffinityTerms in its right place.

**Comment by ahg-g** [comment]
@Huang-Wei can you please take a look again.
